### ISSUE TITLE
Search API

### ISSUE THREAD

**Issue by butonic** [cernbox/reva#41]
AFAICT the cs3 apis do not talk about search.
For now we could implement searching in file metadata and content based on elasticsearch, however I think we need a search api in cs3 as well because the storage itself might have search capabilities. At least optionally.

@labkode what are your thoughts on search?

also see https://github.com/owncloud/nexus/issues/17

**Comment by labkode** [comment]
@butonic Yes, I think the storage provider could provide a find endpoint for this purpose. We just need to be careful on what query filters we allow to not force storage providers to include complicate logic, like "give me all the files where @butonic has written a comment on the 1st of April".

**Comment by butonic** [comment]
also see https://github.com/cernbox/cs3apis/issues/8
